### DISCUSSION TITLE
Using custom myMiddlewares function with TypeScript and slice pattern

### DISCUSSION THREAD

**Discussion by imalgrab** [pmndrs/zustand#2110] [Q&A]
Hey everyone, I'm having a difficult time trying to figure out the proper type for `myMiddlewares` function the same as showcased [here](https://docs.pmnd.rs/zustand/guides/typescript#using-middlewares).
My store is defined using [slices pattern](https://docs.pmnd.rs/zustand/guides/typescript#slices-pattern) like this:

```
const useBoundStore = create<BoundStore>()(
  devtools(
    persist(
      (...initializer) => ({
        ...createFishSlice(...initializer),
        ...createBearSlice(...initializer),
      }),
      persistOptions
    )
  )
);
```

I'd like to use persist middleware conditionally this is why I'm trying to wrap it with `myMiddlewares` function:
```
const myMiddlewares = (f) => {
  if (shouldUsePersist) {
    return devtools(persist(f, persistOptions));
  }

  return devtools(f);
}
```

and finally using it combined:

```
const useBoundStore = create<BoundStore>()(
  myMiddlewares((...initializer) => ({
    ...createFishSlice(...initializer),
    ...createBearSlice(...initializer),
  }))
);
```

Can somebody please assist me with how to type the `f` parameter in myMiddleware function? Thanks in advance!

**Comment by dai-shi** [comment]
> I'd like to use persist middleware conditionally

Theoretically, that means you can't type things statically.

**Reply by imalgrab** [reply]
Okay, any guidance on typing the `myMiddlewares` parameter `f` if not using the middlewares conditionally?

**Reply by dai-shi** [reply]
Like this?
```js
const myMiddlewares = (f) => {
  return devtools(persist(f, persistOptions));
}
```
I think the typescript guide in docs should help. But, it's easier if you can avoid typing custom middleware.
